### PR TITLE
Modal Component

### PR DESCRIPTION
**Pull request by nighto**

### CODE REVIEW

File: src/components/overlays/Modal/Modal.tsx [hunk 1]
```diff
@@ -53,68 +100,111 @@ export type ModalProps = React.PropsWithChildren<{
 /**
  * Modal component.
  */
-export const Modal = ({ children, unstyled, className, active, onClose, closeable = true }: ModalProps) => {
+const Modal = ({
+  children,
+  unstyled,
+  className,
+  size = "medium",
+  closeable = true,
```
Comment (mkrause) [inline src/components/overlays/Modal/Modal.tsx:108]: Please double check whether this also blocks closing using the Escape key.

**Review by mkrause — CHANGES_REQUESTED** [review]
Block Escape key closing the modal when closeable = false.

**Review by mkrause — CHANGES_REQUESTED** [review]

File: src/components/overlays/Modal/Modal.tsx [hunk 2]
```diff
@@ -53,68 +100,127 @@ export type ModalProps = React.PropsWithChildren<{
...
+      }
+    },
+    [close],
+  );
+
+  // prevent closing dialog with Esc key
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e.key !== 'Escape') {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (nighto) [reply src/components/overlays/Modal/Modal.tsx:175]: Yeah, I did try that but unfortunately the `cancel` event is fired *after* it closed the dialog, not before. There's also a bug that if you double press Esc, Chrome would close the dialog nevertheless.

https://stackoverflow.com/questions/61021135/prevent-dialog-from-closing-on-keydown-esc-in-chrome

File: src/components/overlays/Modal/Modal.stories.tsx [hunk 3]
```diff
@@ -76,16 +128,17 @@ const ModalWithSpinnerTrigger = ({ triggerLabel = 'Open modal with spinner (it w
   return (
     <>
       <Button variant="primary" onPress={onPress}>{triggerLabel}</Button>
-      <Modal {...modalProps} active={active} onClose={onClose} closeable={false} className={cl['bk-modal-spinner']}/>
+      <Modal {...modalProps} active={active} onClose={onClose} closeable={false} className={cl['bk-modal-spinner']} size="fullscreen"/>
     </>
   );
 };
 
 export const ModalWithSpinner: Story = {
```
[the remark this answers is not in the document]
Comment (nighto) [reply src/components/overlays/Modal/Modal.stories.tsx:136]: Fixed on latest commit

File: src/components/overlays/Modal/Modal.tsx [hunk 4]
```diff
@@ -53,68 +100,127 @@ export type ModalProps = React.PropsWithChildren<{
...
+      </div>
     </dialog>
   );
 };
+
+Modal.Content = ModalContent;
+Modal.Header = ModalHeader;
+Modal.Footer = ModalFooter;
+
+export { Modal };
```
[the remark this answers is not in the document]
Comment (nighto) [reply src/components/overlays/Modal/Modal.tsx:211]: Fixed

**Review by mkrause — APPROVED** [review]

**Review by mkrause — APPROVED** [review]

**Pull request merged**